From 3f4de04fbed4bd219e39f9ed2fe316e2a82a0f72 Mon Sep 17 00:00:00 2001
From: Paul Handrigan <Paul.Handrigan@cirrus.com>
Date: Thu, 12 Nov 2015 11:30:24 -0600
Subject: [PATCH 505/959] ASoC: cs35l34: Fix analog gain control

Fix analog gain control so it goes from 3dB to 18dB.

Change-Id: I0cb579234a93ab71d181215328bb1f85d48ac3eb
Signed-off-by: Paul Handrigan <Paul.Handrigan@cirrus.com>
Reviewed-on: http://gerrit.mot.com/802195
SME-Granted: SME Approvals Granted
Submit-Approved: Jira Key <jirakey@motorola.com>
Tested-by: Jira Key <jirakey@motorola.com>
SLTApproved: Slta Waiver <sltawvr@motorola.com>
Reviewed-by: Igor Kovalenko <igork@motorola.com>
---
 sound/soc/codecs/cs35l34.c | 8 ++++++++
 sound/soc/codecs/cs35l34.h | 7 -------
 2 files changed, 8 insertions(+), 7 deletions(-)

diff --git a/sound/soc/codecs/cs35l34.c b/sound/soc/codecs/cs35l34.c
index f11343cb617..096425c5c7d 100644
--- a/sound/soc/codecs/cs35l34.c
+++ b/sound/soc/codecs/cs35l34.c
@@ -293,6 +293,14 @@ static int cs35l34_main_amp_event(struct snd_soc_dapm_widget *w,
 }
 
 static DECLARE_TLV_DB_SCALE(dig_vol_tlv, -10200, 50, 0);
+
+static const int gain_ranges[] = {0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 0xA, 0xB,
+				0xC, 0xD, 0xE, 0xF};
+
+static const char * const gain_labels[] = {"3dB", "4dB", "5dB", "6dB", "7dB",
+			"8dB", "9dB", "10dB", "11dB", "12dB", "13dB", "14dB",
+			"15dB", "16dB", "17dB", "18dB"};
+
 static const struct soc_enum amp_gain = SOC_VALUE_ENUM_SINGLE(
 				CS35L34_AMP_ANLG_GAIN_CTL,
 				0, 0xF, ARRAY_SIZE(gain_labels),
diff --git a/sound/soc/codecs/cs35l34.h b/sound/soc/codecs/cs35l34.h
index 0bda307fc86..00343c6df3f 100644
--- a/sound/soc/codecs/cs35l34.h
+++ b/sound/soc/codecs/cs35l34.h
@@ -225,13 +225,6 @@
 /* CS35L34_BST_CTL3 */
 #define BST_IPK_TRIM		0x3F
 
-
-const int gain_ranges[] = {0, 1, 2, 3, 4, 5, 6, 7, 8, 9};
-EXPORT_SYMBOL_GPL(gain_ranges);
-const char *gain_labels[] = {"9db - MIN", "10db", "11db", "12db",
-			"13db", "14db", "15db", "16db", "17db", "18db - MAX"};
-EXPORT_SYMBOL_GPL(gain_labels);
-
 #define CS35L34_RATES (SNDRV_PCM_RATE_48000 | \
 			SNDRV_PCM_RATE_44100 | \
 			SNDRV_PCM_RATE_32000)
-- 
2.11.0

